docs/use-signalr-binding.md

```diff
@@ -0,0 +1,747 @@
+{
+    var userId = req.Query["userid"];
+    var hubName = req.Query["hubname"];
+    var connectionInfo = new SignalRConnectionInfo();
+    var serviceManager = StaticServiceHubContextStore.Get().ServiceManager;
+    connectionInfo.AccessToken = serviceManager
+        .GenerateClientAccessToken(
+            hubName,
+            userId,
+            new List<Claim> { new Claim("claimType", "claimValue") });
+    connectionInfo.Url = serviceManager.GetClientEndpoint(hubName);
+    return connectionInfo;
+}
+```
+
+###### Broadcast messages with output binding
+
+`IServiceHubContext` provides [Microsoft.Azure.SignalR](https://www.nuget.org/packages/Microsoft.Azure.SignalR/)-extended interfaces, if you are familiar with how to use `IHubContext` to send messages in an app server, it will be easy for you to use `IServiceHubContext`.
+
+``` C#
+[FunctionName("broadcast")]
+public static async Task Broadcast(
+    [HttpTrigger(AuthorizationLevel.Anonymous, "post")]HttpRequest req)
+{
+    var message = new JsonSerializer().Deserialize<ChatMessage>(new JsonTextReader(new StreamReader(req.Body)));
+    var serviceHubContext = await StaticServiceHubContextStore.Get().GetAsync("simplechat");
+    await serviceHubContext.Clients.All.SendAsync("newMessage", message);
+}
+```
+
+#### 2.x JavaScript output examples
+
+##### 2.x JavaScript send message output examples
+
+###### Broadcast to all clients
+
+The following example shows a [JavaScript function](https://docs.microsoft.com/azure/azure-functions/functions-reference-node) that sends a message using the output binding to all connected clients. The `Target` is the name of the method to be invoked on each client. The `Arguments` property is an array of zero or more objects to be passed to the client method.
+
+function.json:
+
+```javascript
+{
+  "type": "signalR",
+  "name": "signalRMessages",
+  "hubName": "<hub_name>",
+  "connectionStringSetting": "<name of setting containing SignalR Service connection string>",
+  "direction": "out"
+}
+```
+
+JavaScript code:
+
+```javascript
+module.exports = async function (context, req) {
+    context.bindings.signalRMessages = [{
+        "target": "newMessage",
+        "arguments": [ req.body ]
+    }];
+};
+```
+
+###### Send to a user
+
+You can send a message only to connections that have been authenticated to a user by setting the `userId` property of the SignalR message.
+
+function.json:
+
+```json
+{
+  "type": "signalR",
+  "name": "signalRMessages",
+  "hubName": "<hub_name>",
+  "connectionStringSetting": "<name of setting containing SignalR Service connection string>",
+  "direction": "out"
+}
+```
+
+JavaScript code:
+
+```javascript
+module.exports = async function (context, req) {
+    context.bindings.signalRMessages = [{
+        // message will only be sent to this user ID
+        "userId": "userId1",
+        "target": "newMessage",
+        "arguments": [ req.body ]
+    }];
+};
+```
+
+###### Send to a connection
```

Comment:
shall we take send to a connection as example and simplify send to a user/group with simple dscriptions, most things are duplicated. Besides, shall we also mention that the priority is Connection > User > Group if multiple values are set when send messages?